tests/unit/test_pipeline.py

```diff
@@ -93,3 +95,150 @@ def test_migrate():
     p = Pipeline("tests/data/pipeline", "pipeline-one")
     migrations = p.migrations()
     assert migrations == {"field-one": "FieldOne"}
+
+
+def test_lookups_get_max_entity_success():
+    """
+    test entity num generation functionality
+    :return:
+    """
+
+    pipeline_name = "ancient-woodland"
+    lookups = Lookups("")
+    max_entity_num = lookups.get_max_entity(pipeline_name)
+
+    assert max_entity_num == 0
+
+    entry = {
+        "prefix": "ancient-woodland",
+        "resource": "",
+        "organisation": "government-organisation:D1342",
+        "reference": "1",
+        "entity": "12344",
+    }
+    lookups.entries.append(entry)
+    expected_entity_num = 12344
+
+    assert lookups.get_max_entity(pipeline_name) == expected_entity_num
+
+    max_entity_num = lookups.get_max_entity(pipeline_name)
+    lookups.entity_num_gen.state["current"] = max_entity_num
+    lookups.entity_num_gen.state["range_max"] = max_entity_num + 10
+    expected_entity_num = 12345
+
+    assert lookups.entity_num_gen.next() == expected_entity_num
+
+
+def test_lookups_validate_entry_success():
+    """
+    test csv validate_entry functionality
+    :return:
+    """
+    lookups = Lookups("")
+
+    entry = {
+        "prefix": "ancient-woodland",
+        "resource": "",
+        "organisation": "government-organisation:D1342",
+        "reference": "1",
+        "entity": "",
+    }
+
+    lookups.validate_entry(entry)
+
+
+def test_lookups_validate_entry_failure():
+    """
+    test csv validate_entry functionality for various errors
+    :return:
+    """
+    lookups = Lookups("")
+    invalid_entries = [
```

Comment:
this is better as less duplication you can use 
`@pytest,mark.parametise`
to remove them from inside the test and the test willl run three separate times. This can help when understanding what cause a test to fail. see [here](https://docs.pytest.org/en/7.3.x/how-to/parametrize.html) for some documentation

Reply:
Any thoughts on this?

Reply:
Well I've accepted past comments that I've used too many fixtures and it made the code hard to read, coz you have to go up and down the file.
I guess I figure this has everything in one place. After all, parameterizing (usually) is just a looping mechanism controlled by the Test library.
Typically, I would always go for least complexity.

Reply:
I don't think it makes a huge difference. it would be a simple decorator function and the console output we be improved as it runs each as an individual test. running the below if would difficult to tell which of the invalid entries was causing the test to fail and a dev would likely need to figure out which one it is before debugging that one. 

It's not a massive deal but as a python dev who uses pytest I would expect this to be a prime example of parameterisation is useful.

Reply:
Looks good! it can also be applied to 

`test_lookups_validate_entry_failure`